spec/src/main/asciidoc/metrics.asciidoc

```diff
@@ -0,0 +1,235 @@
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_timeout_execution_duration`
+| Histogram | Nanoseconds
+| Histogram of execution times for the method
+
+|`ft_<name>_timeout_calls_timed_out_total`
+| Counter | None
+| The number of times the method timed out
+
+|`ft_<name>_timeout_calls_not_timed_out_total`
+| Counter | None
+| The number of times the method completed without timing out
+|===
+
+Note that the sum of `ft_<name>_retry_calls_succeeded_not_retried_total`, `ft_<name>_retry_calls_succeeded_retried_total` and `ft_<name>_retry_calls_failed_total` will give the total number of calls for which the Timeout logic was run. This may be larger than the total number of invocations of the method it's also annotated with `@Retry` because the Timeout logic is applied to each Retry attempt, not to the whole method invocation time.
+
+=== Metrics added for `@CircuitBreaker`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_circuitbreaker_calls_succeeded_total`
+| Counter | None
+| Number of calls allowed to run by the circuit breaker that returned successfully
+
+|`ft_<name>_circuitbreaker_calls_failed_total`
+| Counter | None
+| Number of calls allowed to run by the circuit breaker that then failed
+
+|`ft_<name>_circuitbreaker_calls_prevented_total`
+| Counter | None
+| Number of calls prevented from running by an open circuit breaker
+
+|`ft_<name>_circuitbreaker_time_open_total`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in open state
+
+|`ft_<name>_circuitbreaker_time_half_open_total`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in half-open state
+
+|`ft_<name>_circuitbreaker_time_closed_total`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in closed state
+
+|`ft_<name>_circuitbreaker_opened_total`
+| Counter | None
+| Number of times the circuit breaker has moved from closed state to open state
+|===
+
+Note that the sum of `ft_<name>_circuitbreaker_calls_succeeded_total`, `ft_<name>_circuitbreaker_calls_failed_total` and `ft_<name>_circuitbreaker_calls_prevented_total` will give the total number of calls for which the Bulkhead logic was run.
+
+Similarly, the sum of `ft_<name>_circuitbreaker_time_open_total`, `ft_<name>_circuitbreaker_time_half_open_total` and `ft_<name>_circuitbreaker_time_closed_total` will give the total time that the circuit breaker has been active for.
+
+
+=== Metrics added for `@Bulkhead`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft_<name>_bulkhead_concurrent_executions`
+| Gauge<Long> | None
+| Number of currently running executions
+
+|`ft_<name>_bulkhead_calls_accepted_total`
+| Counter | None
+| Number of calls accepted by the bulkhead
+
+|`ft_<name>_bulkhead_calls_rejected_total`
+| Counter | None
+| Number of calls rejected by the bulkhead
+
+|`ft_<name>_bulkhead_execution_duration`
+| Histogram | Nanoseconds
+| Histogram of executions times
```

Comment:
If you have an `@Timeout` and and `@Bulkhead` on the same method, you will end up with two histograms with basically the same contents.  Histograms are expensive (they store the most recent 1000 samples to do computations), so it might be worth having one `ft_<name>_execution_duration` that is used for any FT annotation that requires execution timing.